Venmo/src/main/java/com/braintreepayments/api/venmo/VenmoClient.kt

```diff
@@ -0,0 +1,352 @@
+        }
+    }
+
+    private fun callbackTokenizeSuccess(deepLinkUri: Uri, callback: VenmoTokenizeCallback) {
+        val paymentContextId = parse(deepLinkUri.toString(), "resource_id")
+        val paymentMethodNonce = parse(deepLinkUri.toString(), "payment_method_nonce")
+        val username = parse(deepLinkUri.toString(), "username")
+
+        val isClientTokenAuth = (braintreeClient.authorization is ClientToken)
+        if (paymentContextId != null) {
+
+            venmoApi.createNonceFromPaymentContext(paymentContextId) { nonce: VenmoAccountNonce?, error: Exception? ->
+
+                if (nonce != null) {
+                    isVaultRequest = sharedPrefsWriter.getVenmoVaultOption(
+                        braintreeClient.applicationContext
+                    )
+                    if (isVaultRequest && isClientTokenAuth) {
+                        vaultVenmoAccountNonce(
+                            nonce.string
+                        ) { venmoAccountNonce: VenmoAccountNonce?, vaultError: Exception? ->
+                            if (venmoAccountNonce != null) {
+                                callbackSuccess(
+                                    callback,
+                                    VenmoResult.Success(venmoAccountNonce)
+                                )
+                            } else if (vaultError != null) {
+                                callbackTokenizeFailure(
+                                    callback,
+                                    VenmoResult.Failure(vaultError)
+                                )
+                            }
+                        }
+                    } else {
+                        callbackSuccess(callback, VenmoResult.Success(nonce))
+                    }
+                } else if (error != null) {
+                    callbackTokenizeFailure(callback, VenmoResult.Failure(error))
+                }
+            }
+        } else if (paymentMethodNonce != null && username != null) {
+            isVaultRequest = sharedPrefsWriter.getVenmoVaultOption(
+                braintreeClient.applicationContext
+            )
+
+            if (isVaultRequest && isClientTokenAuth) {
+                vaultVenmoAccountNonce(
+                    paymentMethodNonce
+                ) { venmoAccountNonce: VenmoAccountNonce?, error: Exception? ->
+
+                    if (venmoAccountNonce != null) {
+                        callbackSuccess(callback, VenmoResult.Success(venmoAccountNonce))
+                    } else if (error != null) {
+                        callbackTokenizeFailure(callback, VenmoResult.Failure(error))
+                    }
+                }
+            } else {
+                val venmoAccountNonce = VenmoAccountNonce(
+                    paymentMethodNonce, false, null, null,
+                    null, null, null, username,
+                    null, null
+                )
```

Comment:
This constructor might read a bit better if all arguments are on different lines and if param names are added for each.

Reply:
Updated to put args on different lines - I'm not a fan of comments for param names since Android Studio provides them for you but can add if it would make it more clear

Reply:
Ah ok - added 👍

Reply:
Ah I didn't realize that `VenmoAccountNonce` was a Java class where we can't use named parameters.

Reply:
Ahhh it's not - I misunderstood this comment 🤦‍♀️. Updated to use named params!